From 9361be5611b82d35370701a87d052ccbe031b095 Mon Sep 17 00:00:00 2001
From: Karthikeyan K <karthikk@habana.ai>
Date: Wed, 19 May 2021 13:18:27 +0300
Subject: [PATCH] [SW-42609] rma: update fatal error print as string

The error indication print during an active rma is updated
to print the type of rma as a string instead of a number for
easier understanding.

Change-Id: Ieea5a32acea1b9e0c269cb6276dc3ce772ad9bf0
Signed-off-by: Karthikeyan K <karthikk@habana.ai>
---

diff --git a/gaudi/rma_info.c b/gaudi/rma_info.c
index 7c08163..ec81fa3 100644
--- a/gaudi/rma_info.c
+++ b/gaudi/rma_info.c
@@ -166,6 +166,26 @@
 	return 1;
 }
 
+static void preboot_rma_type_print(struct rma_failure rma_failure)
+{
+	if (rma_failure.sram_ecc)
+		preboot_puts_raw("SRAM ECC ");
+
+	if (rma_failure.hbm_ecc)
+		preboot_puts_raw("HBM ECC ");
+
+	if (rma_failure.hbm_parity)
+		preboot_puts_raw("HBM Parity ");
+
+	if (rma_failure.row_replacement)
+		preboot_puts_raw("Row Replacement ");
+
+	if (rma_failure.hbm_init)
+		preboot_puts_raw("HBM Init ");
+
+	preboot_puts("\n");
+}
+
 /* Read RMA header table */
 static void preboot_process_rma_header(void)
 {
@@ -189,7 +209,8 @@
 		i2csi_update_smbus_alert(rma_header);
 		/* set boot error bit for RMA */
 		cpu_boot_err_clrset(CPU_BOOT_ERR0_DEVICE_UNUSABLE_FAIL, true);
-		preboot_puts_hex("Fatal error:", rma_header->rma_failure.data, 0);
+		preboot_puts_raw("RMA Fatal error(s): ");
+		preboot_rma_type_print(rma_header->rma_failure);
 		i2c_rma_failure = rma_header->rma_failure;
 		if (rma_header->rma_failure.hbm_init)
 			rma_hbm_init_err = rma_header->hbm_init_err;
